zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -1428,7 +1447,10 @@ void onConnected(
         }
 
         void close() {
-            state = States.CLOSED;
+            try {
+                changeZkState(States.CLOSED);
+            } catch (IOException e) {
```

Comment:
Please add a comment
We could anyway spend a line of log, thus case won't happen but if it is happening we won't know

Reply:
Thanks for your suggestion. Accept your suggestion. I add one log here. Please review the latest patch.